src/main/scala/firrtl/transforms/FlattenRegUpdate.scala

```diff
@@ -0,0 +1,97 @@
+// See LICENSE for license details.
+
+package firrtl
+package transforms
+
+import firrtl.ir._
+import firrtl.Mappers._
+import firrtl.Utils._
+
+import scala.collection.mutable
+
+/** Flatten register update
+  *
+  * This transform flattens register updates into a single expression on the rhs of connection to
+  * the register
+  */
+// TODO Preserve source locators
+class FlattenRegUpdate extends Transform {
+  def inputForm = MidForm
+  def outputForm = MidForm
+
+  type Netlist = mutable.HashMap[WrappedExpression, Expression]
+
+  private def buildNetlist(mod: Module): Netlist = {
+    val netlist = new Netlist()
+    def onStmt(stmt: Statement): Statement = {
+      stmt.map(onStmt) match {
+        case Connect(_, lhs, rhs) =>
+          netlist(lhs) = rhs
+        case DefNode(_, nname, rhs) =>
+          netlist(WRef(nname)) = rhs
+        case _: IsInvalid => throwInternalError
+        case _ => // Do nothing
+      }
+      stmt
+    }
+    mod.map(onStmt)
+    netlist
+  }
+
+  private def flattenReg(mod: Module): Module = {
```

Comment:
Why is this private? Can you move this to a companion object and add scaladoc describing the behavior of the function?

Reply:
Yeah I can do that